jsonschema-core.xml

```diff
@@ -1394,87 +1412,179 @@
+                        an implementation MAY normalize a URI ending with an empty fragment by removing
+                        the fragment.  However, schema authors SHOULD NOT rely on this behavior
+                        across implementations.
+                        <cref>
+                            This is primarily allowed because older meta-schemas have an empty
+                            fragment in their $id (or previously, id).  A future draft may outright
+                            forbid even empty fragments in "$id".
+                        </cref>
+                    </t>
+                    <t>
+                        This URI also serves as the base URI for relative URI-references in keywords
+                        within the schema resource, in accordance with
+                        <xref target="RFC3986">RFC 3986 section 5.1.1</xref> regarding base URIs
+                        embedded in content.
+                    </t>
+                    <t>
+                        The presence of "$id" in a subschema indicates that the subschema constitutes
+                        a distinct schema resource within a single schema document.  Furthermore,
+                        in accordance with <xref target="RFC3986">RFC 3986 section 5.1.2</xref>
+                        regarding encapsulating entities, if an "$id" in a subschema is a relative
+                        URI-reference, the base URI for resolving that reference is the URI of
+                        the parent schema resource.
+                    </t>
+                    <t>
+                        If no parent schema object explicitly identifies itself as a resource
+                        with "$id", the base URI is that of the entire document, as established
+                        by the steps given in the <xref target="initial-base">previous section.</xref>
                     </t>
                     <section title="Identifying the root schema">
                         <t>
-                            The root schema of a JSON Schema document SHOULD contain an "$id" keyword with
-                            an <xref target="RFC3986">absolute-URI</xref> (containing a scheme, but no fragment),
-                            or this absolute URI but with an empty fragment.
-                            <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
+                            The root schema of a JSON Schema document SHOULD contain an "$id" keyword
+                            with an <xref target="RFC3986">absolute-URI</xref> (containing a scheme,
+                            but no fragment).
                         </t>
                     </section>
-                    <section title="Changing the base URI within a schema file">
+                    <section title="JSON Pointer fragments and embedded schema resources"
```

Comment:
Reviewed this far...
Excited by this change so far for sure =D
Death to URI shadowing!